lib/utf8utils.c

```diff
@@ -33,7 +33,7 @@ _is_character_unsafe(gunichar uchar, const gchar *unsafe_chars)
   if (!unsafe_chars)
     return FALSE;
 
-  return strchr(unsafe_chars, (gchar) uchar) != NULL;
+  return _strchr_optimized_for_single_char_haystack(unsafe_chars, (gchar) uchar) != NULL;
```

Comment:
We discussed header/non-header choice with @bkil-syslogng, and out opinion is that this function (_strchr_optimized_for_single_char_haystack) should be in this c file, near to its usage.
Or if you found more opportunities of usage, please update the PR with these, but in case you should rename this function e.g.: strchr_optimized_for_single_char_haystack (without underscore)